portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -1,32 +1,80 @@
-// Copyright 2019 Google LLC
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//     https://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Entity;
+import com.google.gson.Gson;
+import com.google.sps.servlets.Comment;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.json.JSONObject;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/**
+* DataServlet class for handling HTTP responses and requests.
+*/
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-
+  /**
+  * Handles HTTP get request. Get all comments from the datastore and send as list json in response.
+  * @param request the request received by servlet.
+  * @param response HTTP response to GET request.
+  */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query(Comment.getDatastoreEntityKey());
```

Comment:
I think you are right, other users of the comment class may want to specify their own datastore entity key. The datastore entity key is not a responsibility of the comment class, will restructure this.

Reply:
I decided to make the key a private static final String in DataServlet as you suggested.
See 95a3459 and 53a27fe

Reply:
This might make more sense as something the class takes in, and then a constant in the file that uses the class.

My reasoning is the same, other users of this class can use different keys.

Reply:
Ok, I changed the entity key to a private instance variable of the DataServlet class. Other users of the class can get and set the entity key for each DataServlet object they create. How does this sound?

1b93687

Reply:
I think an even better way of doing it would be to have it be a file level final variable which your class uses.

But your choice, this seems OK as well.

Reply:
Ah I see. As the objects I store in the datastore become more complicated, I think I will opt for a class or interface of global constants that are needed throughout the project. 

How would you declare a file-level variable in java? I don't know if its possible like it is in C++